pkg/agent/controller/agent.go

```diff
@@ -93,22 +145,30 @@ func (a *Controller) serviceExportToRemoteMcs(obj runtime.Object) runtime.Object
 				"origin-namespace": svc.Namespace,
 			},
 		},
-		Spec: lighthousev1.MultiClusterServiceSpec{
-			Items: []lighthousev1.ClusterServiceInfo{
-				a.newClusterServiceInfo(svc, a.clusterID),
-			},
+	}
+	if op == syncer.Delete {
+		err = a.updateExportedServiceStatus(svcExport, "Service to be exported doesn't exist", lighthousev2a1.ServiceExportInitialized, corev1.ConditionFalse)
+		if err != nil {
+			klog.Errorf("Error updating status on service delete for %#v: %v", svcExport, err)
+		}
+		return mcs, false
+	}
+
+	mcs.Spec = lighthousev1.MultiClusterServiceSpec{
+		Items: []lighthousev1.ClusterServiceInfo{
+			a.newClusterServiceInfo(svc, a.clusterID),
 		},
 	}
 	err = a.updateExportedServiceStatus(svcExport, "Service was successfully synced to the broker",
-		corev1.ConditionTrue)
+		lighthousev2a1.ServiceExportExported, corev1.ConditionTrue)
 	if err != nil {
-		klog.Errorf("Error updating status for %#v: %v", svcExport, err)
+		klog.Errorf("Error updating status on service create/update for %#v: %v", svcExport, err)
 	}
-	return mcs
+	return mcs, false
```

Comment:
Why are you doing this part here? FWIU, you only need to react to Service delete, ie:

 ```
if op != syncer.Delete {
    return nil, false
}

svcExport, err := a.lighthouseClient.LighthouseV2alpha1().ServiceExports(svc.Namespace).Get(svc.Name, 
              metav1.GetOptions{})
// Do error checking

mcs := ...
err = a.updateExportedServiceStatus(...)
// error checking
return mcs, false

```

Reply:
This is for scenario where service is created after serviceExport. If service is not present, I am no longer re-queue in ServiceExport creation. This allows us to react to moment service is created instead of delay due to exp backoff on requeue

Reply:
I think this is adding too much overhead, complication, and duplication to try to optimize an edge/uncommon case. IMO, a max of 30 sec to export the service in this case isn't unreasonable, especially since it's a one-time configuration.  We could also reduce that delay (ie make it configurable in the admiral API).

Reply:
Okay. Will only handle delete for now. We can consider this optimization for later.

Reply:
Can't you do the same processing when the Service itself is created?